magpy/stream.py

```diff
@@ -10419,7 +10419,43 @@ def _read(filename, dataformat=None, headonly=False, **kwargs):
 
     return stream
 
-def saveflags(mylist=None,path=None, overwrite=False):
+def readusgsdata(path_or_url, starttime, endtime, elements, sampling_period):
```

Comment:
Can this be more generic to support requesting chunks of urls based on starttime/endtime, whether usgs or not?